File: C#/Sliding Window/643. Maximum Average Subarray I.cs
/* 643. Maximum Average Subarray I
Difficulty: Easy
Submission Link: https://leetcode.com/problems/maximum-average-subarray-i/submissions/1379178195/
 */

public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        
        double average = 0;

        for(int i = 0; i < k; i++){
            average += nums[i];
        }

        double maxAverage = average/k;

        for(int i = k; i < nums.Length; i++) {
            average += nums[i] - nums[i-k];
            maxAverage = Math.Max(maxAverage, average/k);
        }

        return maxAverage;
    }
}

File: JavaScript/Sliding Window/643. Maximum Average Subarray I.js
/* Maximum Average Subarray I
Difficulty: Easy
Submission Link: https://leetcode.com/problems/maximum-average-subarray-i/submissions/1379177598/ */

const findMaxAverage = function (x, k) {
    let sum = x.slice(0, k).reduce((acc, cur) => acc + cur, 0);
    let avg,
        ans = -Infinity;
    for (let l = 0, r = l + k - 1; l <= x.length - k; l++) {
        avg = (sum / k).toFixed(5);
        sum -= x[l];
        r++;
        sum += x[r];
        ans = Math.max(ans, avg);
    }

    return ans;
};